### PR TITLE
[tests-only][full-ci] Remove null-bytes from the test log before processing

### LINKED ISSUE #336: CLI acceptance test processing is not always correct
Example: https://drone.owncloud.com/owncloud/encryption/2451/108/19
```
  Scenario: user moves their own file to the external storage                                                                             # /var/www/owncloud/testrunner/tests/acceptance/features/cliExternalStorage/filesExternalWebdavOwncloud.feature:270
    Given the administrator has created an external mount point with the following configuration about user "Alice" using the occ command # OccContext::adminHasCreatedAnExternalMountPointWithFollowingConfigUsingTheOccCommand()
      | host                   | %remote_server%    |
      | root                   | TestMnt            |
      | secure                 | false              |
      | user                   | %username%         |
      | password               | %password%         |
      | storage_backend        | owncloud           |
      | mount_point            | TestMountPoint     |
      | authentication_backend | password::password |
    And user "Brian" has been created with default attributes and without skeleton files                                                  # FeatureContext::userHasBeenCreatedWithDefaultAttributesAndWithoutSkeletonFiles()
    And user "Brian" has uploaded file with content "Test content for moving file." to "test.txt"                                         # FeatureContext::userHasUploadedAFileWithContentTo()
    When user "Brian" moves file "/test.txt" to "TestMountPoint/test.txt" using the WebDAV API                                            # FeatureContext::userMovesFileUsingTheAPI()
    Then the HTTP status code should be "201"                                                                                             # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And as "Brian" file "TestMountPoint/test.txt" should exist                                                                            # FeatureContext::asFileOrFolderShouldExist()
    And using server "REMOTE"                                                                                                             # FeatureContext::usingServer()
    And as "Alice" file "TestMnt/test.txt" should exist                                                                                   # FeatureContext::asFileOrFolderShouldExist()
    And the content of file "TestMnt/test.txt" for user "Alice" should be "Test content for moving file."                                 # FeatureContext::contentOfFileForUserShouldBe()
      The downloaded content was expected to be 'Test content for moving file.', but actually is 'HBEGIN:oc_encryption_module:OC_DEFAULT_MODULE:cipher:AES-256-CTR:signed:true:encoding:binary:HEND-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------�䃚s��<Q�-������+i?�L{�00iv00=,/�!,�B��/���x00sig00a76ec4b60938afb6f049f0f70d4ba697d950cfd6be41fd6793bb3737d22d17d5xxx'. HTTP status was 200
      Failed asserting that two strings are equal.
      --- Expected
      +++ Actual
      @@ @@
      -'Test content for moving file.'
```

```
--- Failed scenarios:

    /var/www/owncloud/testrunner/tests/acceptance/features/cliExternalStorage/filesExternalWebdavOwncloud.feature:270

25 scenarios (24 passed, 1 failed)
317 steps (316 passed, 1 failed)
11m30.46s (19.96Mb)
```
The scenario clearly failed in the test run. But the output has lots of special-characters (encrypted content in the file.
The code that checks for failures does not find the failed scenario, and think that it has unexpectedly passed:

```
Checking expected failures for suite cliExternalStorage
Info: Scenario cliExternalStorage/filesExternalWebdavOwncloud.feature:270 was expected to fail but did not fail.
```

```
runsh: There were no unexpected failures.
runsh: Total unexpected passed scenarios throughout the test run:
cliExternalStorage/filesExternalWebdavOwncloud.feature:270
runsh: The following Behat test runs exited with non-zero status:
cliExternalStorage had behat exit status 1
```

Maybe the bash script that checks this stuff gets confused when parsing the test output log, due to some combination(s) of special characters?

Investigate and see if we can get the expected-failures detection to work correctly in cases like this.

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Sometimes the test log will contain null bytes (for now in the Encryption app) due to which `grep` commands fail and the script gives unexpected results.

In this PR, I have removed the null bytes from the test log before processing and also added `-a` options to the grep command 
(`-a : Process a binary file as if it were text`) 

## Related Issue
Fixes https://github.com/owncloud/encryption/issues/336

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by saw-jan** [comment]
Reproduced here: https://drone.owncloud.com/owncloud/encryption/2478/1/19
Fix Tested:

**Comment by phil-davis** [comment]
PHP unit tests will pass after #40101 is merged - needs rebase after that.

**Comment by phil-davis** [comment]
Note: I put some `null` bytes into a text file and then tried to `grep` for text content:
```
$ grep last test.txt 
Binary file test.txt matches
```
So `grep` "works out" that the file looks "binary" and does not actually output the matches.

Then
```
$ grep -a last test.txt 
last line
```

works better (even though the file has null bytes still in it).

And it should be even more reliable with the code here, which filters the null bytes out first.

File: tests/acceptance/run.sh [hunk 1]
```diff
@@ -408,10 +408,17 @@ function run_behat_tests() {
 
 	BEHAT_EXIT_STATUS=${PIPESTATUS[0]}
 
+	# remove nullbytes from the test log
+	TEMP_CONTENT=$(tr < ${TEST_LOG_FILE} -d '\000')
```
Comment (phil-davis) [inline tests/acceptance/run.sh:412]: Note: bash variables do not have an official limit on the size of the value, but they realistically can hold "many" MBs. The test logs sometimes contain thousands of lines of text - for example 100,000 lines * 100 characters = about 10 MB. That sort of size will be handled fine.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**